src/Indice.Features.GovGr/GovGrBancappClient.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Net.Http.Headers;
+using Indice.Features.GovGr.Configuration;
+using Indice.Features.GovGr.Interfaces;
+using Indice.Features.GovGr.Models;
+
+namespace Indice.Features.GovGr;
+
+/// <inheritdoc />
+internal class GovGrBancappClient : IBancappService
+{
+    
+    private const string FunctionFqdnProd = "https://bancapp-api.aade.gr";
+    private const string FunctionFqdnStage = "https://testbancapp-api.aade.gr";
+    private readonly HttpClient _httpClient;
+    private readonly GovGrOptions.BancappOptions _settings;
+
+    internal GovGrBancappClient(
+        HttpClient httpClient,
+        GovGrOptions.BancappOptions settings) {
+        _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+        _settings = settings ?? throw new ArgumentNullException(nameof(settings));
+    }
+    
+    private string FunctionUrl => _settings.IsProduction ? FunctionFqdnProd : FunctionFqdnStage;
+    
+    
+    /// <summary>Upload data to eGov Bancapp</summary>
+    public async Task<BancappGCloudUploadResponse> UploadFile(byte[] fileBytes, string fileName) {
```

Comment:
Use a stream as the base implementation. Then wrap it around an extension method.